### PR TITLE
[XRootD] New Package

### PR DESCRIPTION
**Pull request by lucifer1004**
This package and the previous VDT package are needed for [ROOT](https://github.com/root-project/root) to be built.

### CODE REVIEW

**Comment by lucifer1004** [comment]
@giordano This PR is ready to be merged.

File: X/XRootD/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    ArchiveSource("https://github.com/xrootd/xrootd/archive/refs/tags/v$(version).tar.gz", "56a29c88232f2f384e151b148fcaaa8d8db5c5fdc4615193978c8f4f3a99663c")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir
+mkdir build && cd build
+install_license ../xrootd-*/LICENSE
+cmake -DCMAKE_INSTALL_PREFIX=$prefix -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} -DCMAKE_BUILD_TYPE=Release -DENABLE_PERL=FALSE ../xrootd-*/
+make -j`nproc`
```
Comment (giordano) [inline X/XRootD/build_tarballs.jl:19]:
```suggestion
make -j${nproc}
```

File: X/XRootD/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = expand_cxxstring_abis([
+    Platform("i686", "linux"; libc = "glibc"),
+    Platform("x86_64", "linux"; libc = "glibc"),
+    Platform("aarch64", "linux"; libc = "glibc"),
+    Platform("armv6l", "linux"; call_abi = "eabihf", libc = "glibc"),
+    Platform("armv7l", "linux"; call_abi = "eabihf", libc = "glibc"),
+    Platform("powerpc64le", "linux"; libc = "glibc")
+])
```
Comment (giordano) [inline X/XRootD/build_tarballs.jl:32]:
```suggestion
platforms = expand_cxxstring_abis(supported_platforms(; exclude=p->libc(p) != "glibc"))
```

**Comment by giordano** [comment]
This links to libcrypt.so.1, same problem as #4978.

**Comment by giordano** [comment]
See #5501

**Comment by lucifer1004** [comment]
@giordano Could you help check whether the problem has been solved?

File: X/XRootD/build_tarballs.jl [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    ExecutableProduct("xrdgsitest", :xrdgsitest),
+    ExecutableProduct("cmsd", :cmsd),
+    ExecutableProduct("xrdadler32", :xrdadler32),
+    ExecutableProduct("xrdpinls", :xrdpinls)
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = [
+    Dependency(PackageSpec(name="Libuuid_jll", uuid="38a345b3-de98-5d2b-a5d3-14cd9215e700"))
+    Dependency(PackageSpec(name="libxcrypt_legacy_jll", uuid="5ef642bb-a58b-5208-ae37-583168b2c491"))
```
Comment (giordano) [inline X/XRootD/build_tarballs.jl:94]: This should be used only for libc(p) == "glibc"

File: X/XRootD/build_tarballs.jl [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+mkdir build && cd build
+install_license ../xrootd-*/LICENSE
+cmake -DCMAKE_INSTALL_PREFIX=$prefix -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} -DCMAKE_BUILD_TYPE=Release -DENABLE_PERL=FALSE ../xrootd-*/
+make -j${nproc}
+make install
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = expand_cxxstring_abis(supported_platforms(; exclude=p->libc(p) != "glibc"))
```
Comment (lucifer1004) [inline X/XRootD/build_tarballs.jl:25]: @giordano None-glibc platforms have been excluded here.

Comment (giordano) [reply X/XRootD/build_tarballs.jl:25]: I referred only to the depency libxcrypt_legacy_jll, not the entire recipe

Comment (lucifer1004) [reply X/XRootD/build_tarballs.jl:25]: To clarify, none-glibc platforms were already excluded at first, not after applying lbxcrypt_legacy_jll.

**Pull request merged**